src/code/z_parameter.c

```diff
@@ -94,22 +1086,108 @@
 
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80115844.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80115908.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801159c0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_801159EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/func_80115A14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_parameter/Parameter_AddMagic.s")
+s32 Health_ChangeBy(GlobalContext* globalCtx, s16 healthChange) {
+    if (healthChange > 0) {
+        play_sound(NA_SE_SY_HP_RECOVER);
+    } else if ((gSaveContext.save.playerData.doubleDefense != 0) && (healthChange < 0)) {
+        healthChange >>= 1;
+    }
+
+    gSaveContext.save.playerData.health += healthChange;
+
+    if (((void)0, gSaveContext.save.playerData.health) > ((void)0, gSaveContext.save.playerData.healthCapacity)) {
+        gSaveContext.save.playerData.health = gSaveContext.save.playerData.healthCapacity;
+    }
+
+    if (gSaveContext.save.playerData.health <= 0) {
+        gSaveContext.save.playerData.health = 0;
+        return false;
+    } else {
+        return true;
+    }
+}
+
+void Health_GiveHearts(s16 hearts) {
+    gSaveContext.save.playerData.healthCapacity += hearts * 0x10;
+}
+
+void Rupees_ChangeBy(s16 rupeeChange) {
+    gSaveContext.rupeeAccumulator += rupeeChange;
+}
+
+void Inventory_ChangeAmmo(s16 item, s16 ammoChange) {
```

Comment:
I'm definitely starting to think that "Amount" is better than "Ammo".

Reply:
I'm not convinced of "Amount" just yet since that seems too general maybe, but I'm still thinking about it

Reply:
Agreed. Amount seems too general. I would think it would be something like `ItemAmount`. Personally I think I still like `Ammo` even if it is not quite 100% accurate.

Reply:
I don't think we use `Amount` for anything else, though. `ItemAmount` is a reasonable compromise.

Reply:
I'm not quite convinced of either of those. Ammo seems short and gets the point across, it's the amount of items you contain when you can have multiple items in a specific slot. I know ammo is usually associated with something you attack with (which is why is seems a little awkward to use with beans), but even then "bean" ammo is clear to what it is at the end, the number of beans you have.

Reply:
Ammo sounds fine to me

Reply:
Guess we'll keep it, then.